app/controllers/deployments_controller.rb

```diff
@@ -26,7 +26,17 @@ def new
 
   def create
     if push_notification?
-      application = application_by_repo
+
+      application = if use_application_by_name
+                      application_by_name
+                    else
+                      application_by_repo
+                    end
+
+      if application.nil?
+        return
+      end
```

Comment:
`return if application.nil?` works as well, and is a bit neater in Ruby.

Reply:
I will do that, thanks.